server/integration/testsuite/src/test/java/org/infinispan/server/test/client/rest/RESTAsyncDomainIT.java

```diff
@@ -0,0 +1,86 @@
+package org.infinispan.server.test.client.rest;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.infinispan.arquillian.core.InfinispanResource;
+import org.infinispan.arquillian.core.RemoteInfinispanServer;
+import org.infinispan.server.test.category.RESTClusteredDomain;
+import org.infinispan.server.test.util.ManagementClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.infinispan.server.test.client.rest.RESTHelper.delete;
+import static org.infinispan.server.test.client.rest.RESTHelper.fullPathKey;
+import static org.infinispan.server.test.client.rest.RESTHelper.get;
+import static org.infinispan.server.test.client.rest.RESTHelper.head;
+import static org.infinispan.server.test.client.rest.RESTHelper.put;
+import static org.infinispan.server.test.util.ITestUtils.*;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for the REST client putAsync header.
+ *
+ * @author <a href="mailto:jvilkola@redhat.com">Jozef Vilkolak</a>
+ */
+@RunWith(Arquillian.class)
+@Category({ RESTClusteredDomain.class })
+public class RESTAsyncDomainIT extends AbstractRESTAsyncIT {
+
+    private static final int REST_PORT1 = 8081;
+    private static final int REST_PORT2 = 8231;
+
+    @InfinispanResource(value = "master:server-one", jmxPort = 4447)
+    RemoteInfinispanServer server1;
+
+    @InfinispanResource(value = "master:server-two", jmxPort = 4597)
+    RemoteInfinispanServer server2;
+
+    @BeforeClass
+    public static void beforeClass() {
+        ManagementClient client = ManagementClient.getInstance();
+        try {
+            //do nothing in dist mode, the cache is already there
+            if (isReplicatedMode()) {
+                client.addSocketBinding("rest-repl", "clustered-sockets", REST_PORT1);
```

Comment:
How about creating a generic initialize and cleanup methods in the abstract class (`initDistributedMode`, `initReplicatedMode` etc). This way you would only need to override some of them in your test. I think it would look a bit nicer.